spec.html

```diff
@@ -1898,6 +1873,174 @@ <h1>%TypedArray%.prototype.sort ( _comparefn_ )</h1>
+          1. Repeat, while _k_ &lt; _final_
+            1. Let _Pk_ be ! ToString(_k_).
+            1. Perform ! Set(_O_, _Pk_, _value_, *true*).
+            1. Increase _k_ by 1.
+          1. Return _O_.
+        </emu-alg>
+      </emu-clause>
+
+        <!-- es6num="22.2.3.22.1" -->
+        <emu-clause id="sec-%typedarray%.prototype.set-array-offset">
+          <h1>%TypedArray%.prototype.set ( _array_ [ , _offset_ ] )</h1>
+          <p>Sets multiple values in this _TypedArray_, reading the values from the object _array_. The optional _offset_ value indicates the first element index in this _TypedArray_ where values are written. If omitted, it is assumed to be 0.</p>
+          <emu-alg>
+            1. Assert: _array_ is any ECMAScript language value other than an Object with a [[TypedArrayName]] internal slot. If it is such an Object, the definition in <emu-xref href="#sec-%typedarray%.prototype.set-typedarray-offset"></emu-xref> applies.
+            1. Let _target_ be the *this* value.
+            1. If Type(_target_) is not Object, throw a *TypeError* exception.
+            1. If _target_ does not have a [[TypedArrayName]] internal slot, throw a *TypeError* exception.
+            1. Assert: _target_ has a [[ViewedArrayBuffer]] internal slot.
+            1. Let _targetOffset_ be ? ToInteger(_offset_).
+            1. If _targetOffset_ &lt; 0, throw a *RangeError* exception.
+            1. Let _targetBuffer_ be _target_.[[ViewedArrayBuffer]].
+            1. If IsDetachedBuffer(_targetBuffer_) is *true*, throw a *TypeError* exception.
+            1. Let _targetLength_ be _target_.[[ArrayLength]].
+            1. Let _targetName_ be the String value of _target_.[[TypedArrayName]].
+            1. Let _targetElementSize_ be the Number value of the Element Size value specified in <emu-xref href="#table-49"></emu-xref> for _targetName_.
+            1. Let _targetType_ be the String value of the Element Type value in <emu-xref href="#table-49"></emu-xref> for _targetName_.
+            1. Let _targetByteOffset_ be _target_.[[ByteOffset]].
+            1. Let _src_ be ? ToObject(_array_).
+            1. Let _srcLength_ be ? ToLength(? Get(_src_, `"length"`)).
+            1. If _srcLength_ + _targetOffset_ &gt; _targetLength_, throw a *RangeError* exception.
+            1. Let _targetByteIndex_ be _targetOffset_ &times; _targetElementSize_ + _targetByteOffset_.
+            1. Let _k_ be 0.
+            1. Let _limit_ be _targetByteIndex_ + _targetElementSize_ &times; _srcLength_.
+            1. Repeat, while _targetByteIndex_ &lt; _limit_
+              1. Let _Pk_ be ! ToString(_k_).
+              1. <del>Let _kNumber_ be ? ToNumber(? Get(_src_, _Pk_)).</del>
+              1. <ins>Let _value_ be ? Get(_src_, _Pk_)</ins>
+              1. <ins>If _typedArray_.[[TypedArrayName]] is `"BigUint64Array"` or `"BigInt64Array"`, let _value_ be ? ToBigInt64(_value_)</ins>
```

Comment:
This logic (this line and the next one) seems like it’s repeated; is it perhaps worth defining an internal method on typed arrays to encapsulate and reuse it?